r1cs/gadgets/std/src/groups/group_vec.rs

```diff
@@ -0,0 +1,316 @@
+        }
+        Ok(Self::new(sum_vec))
+    }
+
+    fn double_in_place<CS: ConstraintSystemAbstract<ConstraintF>>(&mut self, mut cs: CS) -> Result<(), SynthesisError> {
+        for (i, el) in self.vec.iter_mut().enumerate() {
+            el.double_in_place(cs.ns(|| format!("double self[{}]", i)))?
+        }
+        Ok(())
+    }
+
+    fn negate<CS: ConstraintSystemAbstract<ConstraintF>>(&self, mut cs: CS) -> Result<Self, SynthesisError> {
+        let res_vec = self.iter().enumerate().map(|(i, el)| {
+            el.negate(cs.ns(|| format!("negate self[{}]", i)))
+        }).collect::<Result<Vec<_>, SynthesisError>>()?;
+        Ok(Self::new(res_vec))
+    }
+
+    fn mul_bits<'a, CS: ConstraintSystemAbstract<ConstraintF>>(&self, mut cs: CS, bits: impl Iterator<Item=&'a Boolean>) -> Result<Self, SynthesisError> {
+        let mut res_vec = Vec::with_capacity(self.len());
+        let bits = bits.cloned().collect::<Vec<_>>();
+        for (i,el) in self.iter().enumerate() {
+            let el_times_bits = el.mul_bits(cs.ns(|| format!("self[{}]*bits", i)), bits.iter())?;
+            res_vec.push(el_times_bits)
+        }
+        Ok(Self::new(res_vec))
+    }
+
+
+    fn mul_bits_fixed_base<CS: ConstraintSystemAbstract<ConstraintF>>(base: &GroupVec<G>, mut cs: CS, bits: &[Boolean]) -> Result<Self, SynthesisError> {
+        let mut res_vec = Vec::with_capacity(base.len());
+        for (i, el) in base.iter().enumerate() {
+            res_vec.push(GG::mul_bits_fixed_base(el, cs.ns(|| format!("mul bits for fixed base {}", i)), bits)?);
+        }
+
+        Ok(Self::new(res_vec))
+    }
+
+
+    fn cost_of_add() -> usize {
+        unimplemented!()
+    }
+
+    fn cost_of_double() -> usize {
+        unimplemented!()
+    }
+
+    fn mul_bits_fixed_base_with_precomputed_base_powers<'a, CS, I, B>(&mut self, _cs: CS, _scalar_bits_with_base_powers: I) -> Result<(), SynthesisError> where CS: ConstraintSystemAbstract<ConstraintF>, I: Iterator<Item=(B, &'a GroupVec<G>)>, B: Borrow<Boolean>, GroupVec<G>: 'a {
+        todo!()
+    }
+
+    fn mul_bits_fixed_base_with_3_bit_signed_digit_precomputed_base_powers<'a, CS, I, J, B>(_cs: CS, _bases: &[B], _powers: &[J]) -> Result<Self, SynthesisError> where CS: ConstraintSystemAbstract<ConstraintF>, I: Borrow<[Boolean]>, J: Borrow<[I]>, B: Borrow<[GroupVec<G>]> {
+        todo!()
+    }
+
+    fn fixed_base_msm_with_precomputed_base_powers<'a, CS, T, I, B>(_cs: CS, _bases: &[B], _scalars: I) -> Result<Self, SynthesisError> where CS: ConstraintSystemAbstract<ConstraintF>, T: 'a + ToBitsGadget<ConstraintF> + ?Sized, I: Iterator<Item=&'a T>, B: Borrow<[GroupVec<G>]> {
+        todo!()
+    }
+
+    fn fixed_base_msm<'a, CS, T, IS, IB>(_cs: CS, _bases: IB, _scalars: IS) -> Result<Self, SynthesisError> where CS: ConstraintSystemAbstract<ConstraintF>, T: 'a + ToBitsGadget<ConstraintF> + ?Sized, IS: Iterator<Item=&'a T>, IB: Iterator<Item=&'a GroupVec<G>> {
```

Comment:
It would be nice if this function were actually implemented. It would be useful for computing the commitment of the input poly inside the Marlin verifier gadget. A starting point for the implementation could be [the code](https://github.com/HorizenOfficial/ginger-lib/blob/31b4a7dc76a95e1e43fbb5c58b2c8b007b847219/proof-systems/src/darlin/marlin/src/constraints/mod.rs#L210) in marlin verifier gadget